NOTIFICATION: Fix console log error and show correct error message after letter proofing failed

eunjuhuss opened the pull request:
#### Description:
not able to see error message when failed letter proofing 
After failed letter proofing an error(can not read property) in console

- before fix
![Screenshot 2021-04-29 at 16 34 19](https://user-images.githubusercontent.com/44289056/116568718-ffb7cc00-a908-11eb-8d31-ce2ac5566dd5.png)

- after fix
![Screenshot 2021-04-29 at 16 33 35](https://user-images.githubusercontent.com/44289056/116568729-034b5300-a909-11eb-9840-158f46b52e86.png)





#### For reviewer:
- [ ] Read the above description
- [ ] Reviewed the code changes
- [ ] Navigate to the branch (pulled the latest version)
- [ ] Run the code and been able to execute the expected function
- [ ] Check any styling on both desktop and mobile sizes

johanlundberg commented on src/notify-middleware.js:
```diff
@@ -2,17 +2,19 @@ import * as actions from "./login/components/Notifications/Notifications_actions
-    if (action.error && action.payload) {
-      console.log("this is action.payload", action.payload);
-      console.log("this is action.error", action.error);
+    if(action.type.startsWith("GET_LETTER")){
```
Can this be solved in the reducer?
https://github.com/SUNET/eduid-front/blob/eunju-fix-notify-error/src/reducers/LetterProofing.js#L42

njons reviewed (commented):
I think I am missing some information about why this is the chosen fix for this problem. Maybe you can help me clarify?

The fix shows an error about the use of an incorrect confirmation code. However the code is sent via action POST_LETTER_PROOFING_VERIFY_CODE and if incorrect returns  POST_LETTER_PROOFING_VERIFY_CODE_FAIL. 
- How does the GET_LETTER code address the failure of the verification code?
- There is a GET_LETTER_PROOFING_PROOFING_SUCCESS, but it is not failing when the verification code is incorrect 

I am also wondering if there is a reason we are not just addressing the undefined nin mentioned in the console error? Often when the redux store is not ready when my components want to load data I do a an `if(state.email.emails !== undefined)`, does that work here?

eunjuhuss commented:
this is two different things

eunjuhuss commented:
Sorry for the insufficient explanation

In this PR I added 2 things.
1. fix issue “not presenting correct error messages in notification bar and error  
[can not read property ‘nin’ of undefined] in console.log

   `else if(action.payload.error && action.payload.error.nin)`
    This code is used for bug fix

2. replacing code 
We had specific exception for presenting messages when the page was loaded. 
When page was loaded, “GET_LETTER_PROOFING”  triggerd and show the error messages. To prevent the bow error we need to get a request for showing correct data in vetting button (by post). Så the new code is preventing the presentation of error messages.
So I have changed:

 `else if (action.payload && action.payload.message && !action.payload.message.includes("letter"))`

  To

```
if(action.type.startsWith("GET_LETTER")){
	delete action.payload.message;
	delete action.payload.error;
}
```

njons commented:
Thank you for the information! Having a generic message on load sounds like a good idea. 
Is there a reason why we choose to not define the nin in question? If there is information from the redux store that is undefined on load would `if(state.nins.nin !== undefined) { // choose what to show only if nin is available }` in the component work?

eunjuhuss commented:
maybe Im wrong, we are talking about two different things.

eunjuhuss commented:
Yes I agree, having a generic message on load is a good. 

Before we changed vetting by post we need GET Requests when we click the button. So we can get correct messages. 

But we changed the layout so we get the information when the page is loaded, so we can display information in the vetting button, ex when letter was sent and when the code is expired. 
Now we need GET Request when page is load.
we need the data but we don’t want to get the message/errors when the page is loaded. 
ex. user has not requested to get a letter but the user get error message (letter.no_state_found) when user is loading the page.

And I think we are talking about different things
`action.payload.error.nin`
This code is for showing detailed error message for user if the user get action.payload.error.nin. it will show relavant error message.

njons commented:
Thanks! Let me know if you want me to merge this or if you'd like Johan to do it in case you want to address his comment.
